### PR TITLE
Stage

### PR DESCRIPTION
**Pull request by bob-isreal**
The Extensions to the original String class in Javascript Has just been completed and it is awaiting review

### CODE REVIEW

**Comment by bob-isreal** [comment]
Screenshot 2019-05-09 at 12 35 03 PM
Screenshot 2019-05-09 at 12 57 14 PM
Attached Are Some ScreenShots Of My Test case

File: main/app.js [hunk 1]
```diff
@@ -1,6 +1,69 @@
-String.prototype.hasVowels = function() {};
-String.prototype.toUpper = function() {};
-String.prototype.toLower = function() {};
-String.prototype.ucFirst = function() {};
+String.prototype.hasVowels = function() {
+    return /[aeiou]/.test(this);
```
Comment (mogbeyi-david) [inline main/app.js:5]: Please add a comment to explain what this regex is matching

File: main/app.js [hunk 2]
```diff
@@ -1,6 +1,69 @@
...
+};
+String.prototype.toUpper = function() {
+    var converted = '';
+    for (var index = 0; index < this.length; index++) {
+        converted += String.fromCharCode(this.charCodeAt(index) & 223);
```
Comment (mogbeyi-david) [inline main/app.js]: Please add a comment here to explain what the number **223** is and how the code is working

File: main/app.js [hunk 3]
```diff
@@ -1,6 +1,69 @@
...
+    }
+    return converted;
+};
+String.prototype.toLower = function() {
+    var converted = '';
+    for (var index = 0; index < this.length; index++) {
+        converted += String.fromCharCode(this.charCodeAt(index) | 32);
```
Comment (mogbeyi-david) [inline main/app.js]: Please add a comment here to explain what this line of code is doing and maybe consider splitting the implementation over two lines to make it more readable

File: main/app.js [hunk 4]
```diff
@@ -1,6 +1,69 @@
...
+    }
+    return converted;
+};
+String.prototype.ucFirst = function() {
+    return /[a-zA-Z]/.exec(this)[0] + this.substring(1).toLower();
```
Comment (mogbeyi-david) [inline main/app.js]: Wow! Please add comments here and split the implementation over several lines. This your one-liner; I hail oooo!!

File: main/app.js [hunk 5]
```diff
@@ -1,6 +1,69 @@
...
+};
 
-module.exports = String.prototype;
+String.prototype.isQuestion = function() {
+    return /\?$/.test(this);
```
Comment (mogbeyi-david) [inline main/app.js:42]: Please add comments to explain what pattern this is matching.

File: main/app.js [hunk 6]
```diff
@@ -1,6 +1,69 @@
...
+};
+String.prototype.words = function() {
+    var pattern = /[a-zA-Z]/g;
```
Comment (mogbeyi-david) [inline main/app.js]: Please add comments to explain what pattern this is matching.

File: main/app.js [hunk 7]
```diff
@@ -1,6 +1,69 @@
...
+    return this.match(pattern);
+};
+String.prototype.wordCount = function() {
+    return this.words().length;
+};
+String.prototype.toCurrency = function() {
+    var digitPattern = /^[0-9]+[.]?[0-9]{1,2}$/;
```
Comment (mogbeyi-david) [inline main/app.js:54]: Please add comments to explain what pattern this is matching.

File: main/app.js [hunk 8]
```diff
@@ -1,6 +1,69 @@
...
+    return digitPattern.test(this)
+        ? Number(this)
+              .toFixed(2)
+              .replace(/\d(?=(\d{3})+\.)/g, '$&,')
```
Comment (mogbeyi-david) [inline main/app.js]: Please add comments to explain what pattern this is matching.

File: main/app.js [hunk 9]
```diff
@@ -1,6 +1,69 @@
...
+    var pattern = /[a-zA-Z]/g;
+    return this.match(pattern);
+};
+String.prototype.wordCount = function() {
+    return this.words().length;
+};
+String.prototype.toCurrency = function() {
+    var digitPattern = /^[0-9]+[.]?[0-9]{1,2}$/;
+    return digitPattern.test(this)
+        ? Number(this)
```
Comment (mogbeyi-david) [inline main/app.js]: I don't think a ternary operator is the best implementation here. It seems to limit readability and the way it spans multiple lines without comments is not really good looking.

File: main/app.js [hunk 10]
```diff
@@ -1,6 +1,69 @@
...
+              .toFixed(2)
+              .replace(/\d(?=(\d{3})+\.)/g, '$&,')
+        : 'Invalid Entry';
+};
+String.prototype.fromCurrency = function() {
+    var digit = /(\d+,)+/;
```
Comment (mogbeyi-david) [inline main/app.js:67]: Please add comments to explain what pattern this is matching.

File: main/app.js [hunk 11]
```diff
@@ -1,6 +1,69 @@
...
+    return digit.test(this) ? this.replace(/,(?=\d)/g, '') : 'Invalid Entry';
```
Comment (mogbeyi-david) [inline main/app.js:69]: Please add comments to explain what pattern this is matching.

**Review by mogbeyi-david — CHANGES_REQUESTED** [review]
The logical implementation of the code is really good, but the cleanliness can be much better. Did you attend the class on clean codes? I do not see the implementation of the basic concepts taught in that class here. Please review the changes as specified and get back to me.

**Pull request closed without merging**